pkg/config/types/v1alpha1/cluster_config.go

```diff
@@ -143,212 +117,25 @@ func (e *EtcdConfig) CA() *x509.PEMEncodedCertificateAndKey {
-		return "", err
-	}
-
-	encodedObjs := [][]byte{}
-	for _, obj := range []kuberuntime.Object{initConfig, clusterConfig} {
-		encoded, err := kubeadmutil.MarshalToYamlForCodecs(obj, kubeadmv1beta2.SchemeGroupVersion, kubeadmscheme.Codecs)
-		if err != nil {
-			return "", err
-		}
-		encodedObjs = append(encodedObjs, encoded)
-	}
-
-	encoded, err := kubeadmutil.MarshalToYamlForCodecs(kubeletConfig, kubeletconfigv1beta1scheme.SchemeGroupVersion, kubeadmscheme.Codecs)
-	if err != nil {
-		return "", err
-	}
-	encodedObjs = append(encodedObjs, encoded)
-
-	encoded, err = kubeadmutil.MarshalToYamlForCodecs(proxyConfig, kubeproxyconfigv1alpha1scheme.SchemeGroupVersion, kubeadmscheme.Codecs)
-	if err != nil {
-		return "", err
-	}
-	encodedObjs = append(encodedObjs, encoded)
-
-	kubeadmConfig := bytes.Join(encodedObjs, []byte("---\n"))
-
-	return string(kubeadmConfig), nil
+// Token implements the Configurator interface.
+func (c *ClusterConfig) Token() cluster.Token {
+	return c
 }
 
-func (c *ClusterConfig) controlplaneConfig() (string, error) {
-	controlPlaneConfig := &kubeadm.JoinConfiguration{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "JoinConfiguration",
-			APIVersion: "kubeadm.k8s.io/v1beta2",
-		},
-		ControlPlane: &kubeadm.JoinControlPlane{
-			CertificateKey: c.CertificateKey,
-		},
-		Discovery: kubeadm.Discovery{
-			BootstrapToken: &kubeadm.BootstrapTokenDiscovery{
-				Token:                    c.Token,
-				APIServerEndpoint:        c.ControlPlane.IPs[0] + ":443",
-				UnsafeSkipCAVerification: true,
-			},
-		},
-		NodeRegistration: kubeadm.NodeRegistrationOptions{
-			CRISocket: constants.ContainerdAddress,
-			// KubeletExtraArgs: ,
-		},
-	}
-
-	encodedObjs := [][]byte{}
-	encoded, err := kubeadmutil.MarshalToYamlForCodecs(controlPlaneConfig, kubeadmv1beta2.SchemeGroupVersion, kubeadmscheme.Codecs)
-	if err != nil {
-		return "", err
+// ID implements the Configurator interface.
+func (c *ClusterConfig) ID() string {
+	parts := strings.Split(c.BootstrapToken, ".")
+	if len(parts) != 2 {
+		return ""
 	}
-	encodedObjs = append(encodedObjs, encoded)
-
-	kubeadmConfig := bytes.Join(encodedObjs, []byte("---\n"))
-
-	return string(kubeadmConfig), nil
+	return parts[0]
 }
 
-func (c *ClusterConfig) joinConfig() (string, error) {
-	joinConfig := &kubeadm.JoinConfiguration{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "JoinConfiguration",
-			APIVersion: "kubeadm.k8s.io/v1beta2",
-		},
-		Discovery: kubeadm.Discovery{
-			BootstrapToken: &kubeadm.BootstrapTokenDiscovery{
-				Token:                    c.Token,
-				APIServerEndpoint:        c.ControlPlane.IPs[0] + ":443",
-				UnsafeSkipCAVerification: true,
-			},
-		},
-		NodeRegistration: kubeadm.NodeRegistrationOptions{
-			CRISocket: constants.ContainerdAddress,
-			// KubeletExtraArgs: ,
-		},
-	}
-
-	if err := cis.EnforceWorkerRequirements(joinConfig); err != nil {
-		return "", err
+// Secret implements the Configurator interface.
+func (c *ClusterConfig) Secret() string {
```

Comment:
It seems very odd that ClusterConfig.Secret() returns the cluster's node bootstrap secret

Reply:
This uses the nomenclature from Kubernetes. A bootstrap token has and id and a secret portion. Separated by the `.`.